src/components/ConnectedHeaderbar.jsx

```diff
@@ -51,8 +51,7 @@ export function ConnectedHeaderBar({ appsInfoQuery }) {
 
     const appName = useMemo(() => {
         if (!params.appName || !appsInfoQuery.data) {
-            // `undefined` defaults to app title in header bar component, i.e. "Global Shell"
-            return ' '
+            return
```

Comment:
This change still avoids the "Global Shell" flash in the header bar, but also avoids setting " | DHIS2" as the document title if an app isn't found